File: core/src/testFixtures/resources/fixtures/repos/gradle-small/src/test/kotlin/pay/RouterTest.kt
package pay

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource
import pay.handlers.handleUser

// `smoke` also exists in pay.handlers.HandlerTest: two classes in two
// packages, one method name, two distinct namespaced identities (D-27, IX-13).
class RouterTest {
    private val router = Router(mapOf("/user" to ::handleUser))

    @Test
    fun smoke() {
        assertEquals(200, router.route(Request("/user", userId = 7)).status)
    }

    @Test
    fun `unknown path is 404`() {
        assertEquals(404, router.route(Request("/nope")).status)
    }

    @ParameterizedTest
    @ValueSource(strings = ["EUR", "Usd", " gbp "])
    fun normalizesCurrency(raw: String) {
        assertEquals(raw.trim().lowercase(), normalizeCurrency(raw))
    }
}

File: core/src/testFixtures/resources/fixtures/repos/ts-small/test/router.test.ts
import { test } from "node:test";
import assert from "node:assert/strict";
import { Router, normalizeCurrency } from "../src/router.ts";
import type { Request, Response } from "../src/router.ts";

// `smoke` also exists in test/index.test.ts: two files, one test name, two
// distinct namespaced identities (D-27, IX-13). The junit reporter attributes
// each one to its file; the spec and tap reporters do not.

const ok: (req: Request) => Response = (req) => ({ status: 200, body: req.path });

test("smoke", () => {
  assert.equal(new Router({ "/user": ok }).route({ path: "/user" }).status, 200);
});

test("unknown path is 404", () => {
  assert.equal(new Router({ "/user": ok }).route({ path: "/nope" }).status, 404);
});

const currencyCases: ReadonlyArray<readonly [string, string]> = [
  ["EUR", "eur"],
  ["Usd", "usd"],
  [" gbp ", "gbp"],
];

for (const [raw, expected] of currencyCases) {
  test(`normalizeCurrency ${expected}`, () => {
    assert.equal(normalizeCurrency(raw), expected);
  });
}
